chain/network/src/peer/peer_actor.rs

```diff
@@ -1169,9 +1163,11 @@ impl Handler<PeerManagerRequestWithContext> for PeerActor {
     }
 }
 
+type InboundHandshakePermit = tokio::sync::OwnedSemaphorePermit;
```

Comment:
:heart: "we should lean more on tokio-provided synchronization & flow-control patterns" is along-standing wish of mine